include/mtxclient/http/client.hpp

```diff
@@ -446,6 +448,12 @@ class Client : public std::enable_shared_from_this<Client>
           const std::map<mtx::identifiers::User, std::map<std::string, EventContent>> &messages,
           ErrCallback callback);
 
+        //! POST a new room listing to the public rooms directory.
+        void post_public_rooms(const std::string &server, const nhlomann::json &j, 
+                                Callback<mtx::responses::PublicRooms> cb); 
+        //! GET the public rooms directory listing.
+        void get_public_rooms(int limit, const std::string &since, const std::string &server,
```

Comment:
I'd order them the other way around, server is most important, then maybe limit then since? Also add a default value for all of them, since those are optional.